### PR TITLE
ANDROAPP-5532-mobile-ui-Create-Input-Yes-no-Field-component

### PR DESCRIPTION
**Pull request by siddh1004**

### CODE REVIEW

**Review by Balcan — APPROVED** [review]

**Comment by andresmr** [comment]
please, solve conflicts and update the branch

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputYesNoField.kt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * @param legendData manages the legendComponent
+ * @param isRequired controls whether the field is mandatory or not
+ * @param itemSelected controls which item is selected.
+ * @param onItemChange is a callback to notify which item has changed into the block.
+ */
+@Composable
+fun InputYesNoField(
+    title: String,
+    modifier: Modifier = Modifier,
+    orientation: Orientation = Orientation.HORIZONTAL,
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputYesNoField.kt:31]: I think that in the yes/no field the orientation is always horizontal, so we don't need to expose it.

**Review by andresmr — CHANGES_REQUESTED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputYesNoField.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    onClick = {
+                        onItemChange.invoke(null)
+                    },
+                )
+            }
+        },
+    )
+}
+
+enum class InputYesNoFieldValues(val value: String) {
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputYesNoField.kt:94]: this values should be taken from string resources in order to make translations in the future.

**Review by andresmr — CHANGES_REQUESTED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputYesNoField.kt (hunk 2 above)
Comment (siddh1004) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputYesNoField.kt:94]: @andresmr The value already uses string resources for the translations.
` provideStringResource(it.value.lowercase(Locale.getDefault()))`

**Review by andresmr — APPROVED** [review]

**Pull request merged**